lib/dataloader/source.ex

```diff
@@ -42,4 +42,10 @@ defprotocol Dataloader.Source do
   """
   @spec timeout(t) :: number
   def timeout(source)
+
+  @doc """
+  Returns wether or not the source should be running synchronously or asynchronously
+  """
+  @spec async?(t) :: boolean
+  def async?(source)
```

Comment:
So my only concern here is that this is functionally a breaking change right? Any other 3rd party protocol impl will error now that this is here.

Reply:
Yeah, absolutely. that is the main reason this was opened as a draft actually. There may be a way around it, which would cause them simply to warn instead of raise, which would be to do something like:

```elixir
if function_exported?(Dataloader.impl_for(module), :async?, 1) do
  ....
end
```

But that also adds a concrete performance overhead for other users. To prevent breaking this in the future again, if additional options are needed, we could make it `options/1` instead of `async?/1` that way if more options are added we don't need a major release.

```elixir
def options(dataloader) do
  [
    async?: false
  ]
end
```